lib/ecto/adapters/tds.ex

```diff
@@ -183,7 +183,7 @@ defmodule Ecto.Adapters.Tds do
       {:ok, _} ->
         :ok
 
-      {:error, %Tds.Error{mssql: %{number: 1801}}} ->
+      {:error, %{mssql: %{number: 1801}}} ->
```

Comment:
```ex
{:error, %{__struct__: Tds.Error, mssql: %{number: 1801}}} ->
```

Maybe this to somewhat retain the previous intention?

Reply:
I think it wouldn't hurt to add it but since we precisely control where the errors are coming form (the tds driver in this case) I think this is fairly obvious and I personally wouldn't change it. :)